cluster-autoscaler/cloudprovider/clusterapi/README.md

```diff
@@ -66,30 +66,80 @@ To specify the kubeconfig path for the management cluster to monitor, use the
 `--cloud-config` option is not specified it will fall back to using the kubeconfig
 that was provided with the `--kubeconfig` option.
 
+### Autoscaler running in a joined cluster using service account credentials
+```
++-----------------+
+| mgmt / workload |
+| --------------- |
+|    autoscaler   |
++-----------------+
+```
 Use in-cluster config for both management and workload cluster:
 ```
 cluster-autoscaler --cloud-provider=clusterapi
 ```
 
+### Autoscaler running in workload cluster using service account credentials, with separate management cluster
+```
++--------+              +------------+
+|  mgmt  |              |  workload  |
+|        | cloud-config | ---------- |
+|        |<-------------+ autoscaler |
++--------+              +------------+
+```
+
 Use in-cluster config for workload cluster, specify kubeconfig for management cluster:
 ```
-cluster-autoscaler --cloud-provider=clusterapi --cloud-config=/mnt/kubeconfig
+cluster-autoscaler --cloud-provider=clusterapi \
+                   --cloud-config=/mnt/kubeconfig
+```
+
+### Autoscaler running in management cluster using service account credentials, with separate workload cluster
+```
++------------+             +----------+
+|    mgmt    |             | workload |
+| ---------- | kubeconfig  |          |
+| autoscaler +------------>|          |
++------------+             +----------+
 ```
 
 Use in-cluster config for management cluster, specify kubeconfig for workload cluster:
 ```
-cluster-autoscaler --cloud-provider=clusterapi --kubeconfig=/mnt/kubeconfig --clusterapi-cloud-config-authoritative
+cluster-autoscaler --cloud-provider=clusterapi \
+                   --kubeconfig=/mnt/kubeconfig \
+                   --clusterapi-cloud-config-authoritative
+```
+
+### Autoscaler running anywhere, with separate kubeconfigs for management and workload clusters
+```
++--------+               +------------+             +----------+
+|  mgmt  |               |     ?      |             | workload |
+|        |  cloud-config | ---------- | kubeconfig  |          |
+|        |<--------------+ autoscaler +------------>|          |
++--------+               +------------+             +----------+
 ```
 
 Use separate kubeconfigs for both management and workload cluster:
 ```
-cluster-autoscaler --cloud-provider=clusterapi --kubeconfig=/mnt/workload.kubeconfig --cloud-config=/mnt/management.kubeconfig
+cluster-autoscaler --cloud-provider=clusterapi \
+                   --kubeconfig=/mnt/workload.kubeconfig \
+                   --cloud-config=/mnt/management.kubeconfig
+```
+
+### Autoscaler running anywhere, with a common kubeconfig for management and workload clusters
+```
++--------+             +------------+             +----------+
```

Comment:
if they share the same kubeconfig I'd expect this to look like the box in https://github.com/kubernetes/autoscaler/pull/4474/files#diff-eabe434726684600ca7ec154b4f511caa551401fd78268f48a70a812f2ec8040R69 for mgmt / workload and have another box for the autoscaler.

Reply:
yeah, i went back and forth about the designs, i wanted to illustrate that they don't necessarily have to be the same, but i could make it more like the diagram above if you think that makes more sense?